package.json

```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@snowplow/webview-tracker",
+  "version": "0.1.0",
+  "description": "WebView tracker for Snowplow",
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/snowplow-incubator/snowplow-webview-tracker"
+  },
+  "contributors": [
+    "Matus Tomlein"
+  ],
+  "main": "dist/index.js",
+  "typings": "dist/index.d.ts",
+  "files": [
+    "dist",
+    "src"
+  ],
+  "engines": {
+    "node": ">=10"
+  },
+  "scripts": {
+    "start": "tsdx watch",
+    "build": "tsdx build",
```

Comment:
Looking at the docs for tsdx, you should be able to build a `umd` version, which will give you the <script> version of this tracker pretty easily.

That might be good enough for v0.1(?).

However, it's typically good practise for analytics libraries that are loading with <script> to inject a global array before they load, like the JS Tracker does. This ensures any events which are tracked before the library loads are no lost. I imagine this is still the same issue with the webview tracker, so it'd probably be good to create a tag snippet which loads the tracker (which means building a `iife` version). Since tsdx uses rollup under the hood, hopefully you'll be able to get that working.

You'd basically just have different entry points for the iife version and the esm version.

Side note: Looks like tsdx builds cjs versions too? I imagine those are a waste of time these days, unless you're building for node too, which you aren't.

Reply:
Good points, I wasn't very familiar with the JavaScript build formats and learning a bit deeper I realized that tsdx was not the right choice here. It seems to be primarily designed for Node apps and although it is built on rollup and support for iife could be theoretically added, it would require messing in its internals (also it seems to be not well maintained).

So I decided to replace it with [microbundle](https://github.com/developit/microbundle) which is a lightweight builder that uses rollup under the hood and provides better options for browser libraries. It also has support for iife. Another nice feature is that it support the modern build format which runs in modern browsers without being transpiled.

`npm run build` builds the default microbundle formats. I also added `npm run build-iife` which builds an `sp.js` file using the `src/tag.ts` source file. This can be loaded in <script> tags using the same snippet that we use for the JavaScript tracker (it uses the same queue format to receive messages).